src/HomeScreen/hooks/useBaniList.js

```diff
@@ -1,23 +1,33 @@
-import { useEffect, useState } from "react";
-import { useSelector } from "react-redux";
-import FallBack from "../../common/components/FallbackComponent";
-import errorHandler from "../../common/errHandler";
-import { getBaniList } from "../../database/db";
+import { useEffect, useState, useRef } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { FallBack, errorHandler, orderedBani, actions } from "@common";
+import { getBaniList } from "@database";
 
 const useBaniList = () => {
+  const baniList = useSelector((state) => state.baniList);
+  const baniOrder = useSelector((state) => state.baniOrder);
   const [baniListData, setBaniListData] = useState([]);
   const transliterationLanguage = useSelector((state) => state.transliterationLanguage);
+  const prevLanguageRef = useRef(transliterationLanguage);
+  const dispatch = useDispatch();
   useEffect(() => {
-    (async () => {
+    const fetchBaniList = async () => {
```

Comment:
Instead of defining a function inside `useEffect`, we can have the `useCallback` to memorize the function definition, with the side effects it need. and then call it inside the useEffect.

Reply:
Fixed